Meadowlark-js/backends/meadowlark-mongodb-backend/src/repository/Update.ts

```diff
@@ -8,80 +8,117 @@ import { Logger } from '@edfi/meadowlark-utilities';
+    if (resourceInfo.allowIdentityUpdates) {
+      const deleteResult = await deleteDocumentByIdTransaction(
+        { documentUuid, resourceInfo, security, validate: true, traceId },
+        client,
+        session,
+      );
+      // if the document was deleted, it should insert the new version.
+      if (deleteResult.response === 'DELETE_SUCCESS') {
+        // insert the updated document.
+        const upsertResult = await upsertDocumentTransaction(
+          {
+            resourceInfo,
+            documentInfo,
+            documentUuidInserted: documentUuid,
+            meadowlarkId,
+            edfiDoc,
+            validate,
             traceId,
-          );
-
-          const referringDocuments: WithId<MeadowlarkDocument>[] = await mongoCollection
-            .find(onlyDocumentsReferencing([id]), limitFive(session))
-            .toArray();
-
-          const blockingDocuments: BlockingDocument[] = referringDocuments.map((document) => ({
-            // eslint-disable-next-line no-underscore-dangle
-            documentId: document._id,
-            resourceName: document.resourceName,
-            projectName: document.projectName,
-            resourceVersion: document.resourceVersion,
-          }));
-
-          updateResult = {
-            response: 'UPDATE_FAILURE_REFERENCE',
-            failureMessage: { message: 'Reference validation failed', failures },
-            blockingDocuments,
-          };
-
-          await session.abortTransaction();
-          return;
+            security,
+          },
+          client,
+          session,
+        );
+        if (upsertResult.response === 'INSERT_SUCCESS') {
+          updateResult = { response: 'UPDATE_SUCCESS' };
+        } else {
+          updateResult = { response: 'UPDATE_FAILURE_NOT_EXISTS' };
         }
+      } else if (deleteResult.response === 'DELETE_FAILURE_NOT_EXISTS') {
+        updateResult = { response: 'UPDATE_FAILURE_NOT_EXISTS' };
       }
-
-      const document: MeadowlarkDocument = meadowlarkDocumentFrom(
-        resourceInfo,
-        documentInfo,
-        id,
-        edfiDoc,
-        validate,
-        security.clientId,
-      );
-
+      return updateResult;
+    }
+    // Start update transaction
+    await session.withTransaction(async () => {
       await writeLockReferencedDocuments(mongoCollection, document.outboundRefs, session);
 
       // Perform the document update
-      Logger.debug(`${moduleName}.updateDocumentById: Updating document id ${id}`, traceId);
-
-      const { acknowledged, matchedCount } = await mongoCollection.replaceOne({ _id: id }, document, { session });
+      Logger.debug(`${moduleName}.updateDocumentById: Updating document uuid ${documentUuid}`, traceId);
+      const { acknowledged, matchedCount } = await mongoCollection.replaceOne(
+        { _id: meadowlarkId, documentUuid },
+        document,
+        {
+          session,
+        },
+      );
```

Comment:
I think I'd prefer not to have the getById happen in the handler because some backends might be able to optimize the check themselves and avoid a second query.

Reply:
It ended up that ordering can be optimized if it stays in the backend.